zebra/interface.c

```diff
@@ -1567,56 +1567,40 @@ struct cmd_node interface_node = {INTERFACE_NODE, "%s(config-if)# ", 1};
 #endif
 /* Show all interfaces to vty. */
 DEFPY(show_interface, show_interface_cmd,
-      "show interface [vrf NAME$vrf_name] [brief$brief]",
+      "show interface [{vrf <NAME$vrf_name|all$vrf_all>|brief$brief}]",
       SHOW_STR
       "Interface status and configuration\n"
-      VRF_CMD_HELP_STR
+      VRF_FULL_CMD_HELP_STR
       "Interface status and configuration summary\n")
 {
 	struct vrf *vrf;
 	struct interface *ifp;
-	vrf_id_t vrf_id = VRF_DEFAULT;
 
 	interface_update_stats();
 
-	if (vrf_name)
-		VRF_GET_ID(vrf_id, vrf_name, false);
-
-	/* All interface print. */
-	vrf = vrf_lookup_by_id(vrf_id);
-	if (brief) {
-		ifs_dump_brief_vty(vty, vrf);
-	} else {
-		FOR_ALL_INTERFACES (vrf, ifp) {
-			if_dump_vty(vty, ifp);
+	if (vrf_all) {
+		RB_FOREACH (vrf, vrf_name_head, &vrfs_by_name) {
+			if (brief) {
+				ifs_dump_brief_vty(vty, vrf);
+			} else {
+				FOR_ALL_INTERFACES (vrf, ifp)
+					if_dump_vty(vty, ifp);
+			}
 		}
-	}
-
-	return CMD_SUCCESS;
-}
-
-
-/* Show all interfaces to vty. */
-DEFPY (show_interface_vrf_all,
-       show_interface_vrf_all_cmd,
-       "show interface vrf all [brief$brief]",
-       SHOW_STR
-       "Interface status and configuration\n"
-       VRF_ALL_CMD_HELP_STR
-       "Interface status and configuration summary\n")
-{
-	struct vrf *vrf;
-	struct interface *ifp;
+	} else {
+		vrf_id_t vrf_id = VRF_DEFAULT;
 
-	interface_update_stats();
+		if (vrf_name)
+			VRF_GET_ID(vrf_id, vrf_name, false);
```

Comment:
Suggestion:
Why dont we use vrf_lookup_by_name to get vrf pointer? (default case VRF_DEFAULT_NAME)
getting id and then getting vrf ptr from id leads to double look up(except default case).
If this is just to make use of some failure print(vty_out) inside VRF_GET_ID, imo better will write new macro to get vrf pointer and print these warning message in case of failure.

Reply:
@idryzhov I think you still need to address this comment?

Reply:
@sworleys I am not sure that this PR is going to be merged. As @donaldsharp said, FRR expects "vrf NAME" to be right after the main part of the command and this PR breaks it. I've already submitted another PR https://github.com/FRRouting/frr/pull/5301 with a different approach.

Reply:
@sworleys actually I am happy with the approach in this PR as well. Can you, please, make a decision inside Cumulus? I'll address the comment if we're going with this PR.